src/modules/flow/http-server/http-server.c

```diff
@@ -255,40 +248,84 @@ common_response_cb(void *data, struct sol_http_request *request)
         }
     }
 
-    response.response_code = SOL_HTTP_STATUS_INTERNAL_SERVER_ERROR;
-
     if (send_json) {
-        r = sol_buffer_append_printf(&response.content, "{\"%s\":", mdata->path);
-        SOL_INT_CHECK_GOTO(r, < 0, end);
+        r = sol_buffer_append_printf(&response->content,
+            "{\"%s\":", mdata->path);
+        SOL_INT_CHECK_GOTO(r, < 0, err_exit);
     }
 
-    r = type->response_cb(mdata, &response.content, send_json);
-    SOL_INT_CHECK_GOTO(r, < 0, end);
+    r = type->response_cb(mdata, &response->content, send_json);
+    SOL_INT_CHECK_GOTO(r, < 0, err_exit);
 
     if (send_json) {
-        r = sol_buffer_append_char(&response.content, '}');
-        SOL_INT_CHECK_GOTO(r, < 0, end);
+        r = sol_buffer_append_char(&response->content, '}');
+        SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+    }
+
+    err_r = sol_http_param_add(&response->param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, (send_json) ? HTTP_HEADER_CONTENT_TYPE_JSON :
+        HTTP_HEADER_CONTENT_TYPE_TEXT));
+
+    if (!err_r) {
+        r = -ENOMEM;
+        SOL_WRN("Could not set the Content-Type");
+        goto err_exit;
+    }
+
+    return 0;
+
+err_exit:
+    return r;
+}
+
+static int
+common_response_cb(void *data, struct sol_http_request *request)
+{
+    int r = 0;
+    enum sol_http_method method;
+    struct sol_flow_node *node = data;
+    struct http_data *mdata = sol_flow_node_get_private_data(node);
+    const struct http_server_node_type *type;
+    bool updated = false;
+    struct sol_http_response response = {
+        SOL_SET_API_VERSION(.api_version = SOL_HTTP_RESPONSE_API_VERSION, )
+        .content = SOL_BUFFER_INIT_EMPTY,
+        .param = SOL_HTTP_REQUEST_PARAMS_INIT,
+        .response_code = SOL_HTTP_STATUS_INTERNAL_SERVER_ERROR
+    };
+
+    type = (const struct http_server_node_type *)
+        sol_flow_node_get_type(node);
+
+    method = sol_http_request_get_method(request);
+    response.url = sol_http_request_get_url(request);
+
+    if (!is_method_allowed(mdata->allowed_methods, method)) {
+        SOL_INF("HTTP Method not allowed. Method: %d", (int)method);
+        response.response_code = SOL_HTTP_STATUS_FORBIDDEN;
+        r = sol_http_server_send_response(request, &response);
+        if (r < 0)
+            SOL_WRN("Could not send the forbidden response");
```

Comment:
should send an error packet here as in the default error treatment (see label end)